### DISCUSSION TITLE
How is the help menu ordered?

### DISCUSSION THREAD

**Discussion by JeremyLaurenson** [HomeSpan/HomeSpan#867] [Q&A]
I have added a few SpanUserCommand's to my system and when I issue a ? command they seem to be listed in random order.

How are they sorted?

**Comment by JeremyLaurenson** [comment]
I see the code iterates over the user commands using this:

```
      for(auto uCom=homeSpan.UserCommands.begin(); uCom!=homeSpan.UserCommands.end(); uCom++)      // loop over all UserCommands using an iterator
        LOG0("  @%c %s\n",uCom->first,uCom->second->s);

```

But I do not know how UserCommands may be sorted

**Reply by HomeSpan** [reply]
I used a C++ unordered map container as the structure.  There is no well-defined order to how they are stored as the system builds a hash tree automatically.
